src/components/views/rooms/MessageComposerInput.js

```diff
@@ -36,49 +38,30 @@ var sdk = require('../../../index');
 var dis = require("../../../dispatcher");
 var KeyCode = require("../../../KeyCode");
 
+import {contentStateToHTML, HTMLtoContentState, getScopedDecorator} from '../../../RichText';
+
 var TYPING_USER_TIMEOUT = 10000;
 var TYPING_SERVER_TIMEOUT = 30000;
 var MARKDOWN_ENABLED = true;
 
-function mdownToHtml(mdown) {
-    var html = marked(mdown) || "";
-    html = html.trim();
-    // strip start and end <p> tags else you get 'orrible spacing
-    if (html.indexOf("<p>") === 0) {
-        html = html.substring("<p>".length);
-    }
-    if (html.lastIndexOf("</p>") === (html.length - "</p>".length)) {
-        html = html.substring(0, html.length - "</p>".length);
-    }
-    return html;
-}
-
 /*
  * The textInput part of the MessageComposer
  */
-module.exports = React.createClass({
-    displayName: 'MessageComposerInput',
-
-    statics: {
-        // the height we limit the composer to
-        MAX_HEIGHT: 100,
-    },
-
-    propTypes: {
-        tabComplete: React.PropTypes.any,
-
-        // a callback which is called when the height of the composer is
-        // changed due to a change in content.
-        onResize: React.PropTypes.func,
-
-        // js-sdk Room object
-        room: React.PropTypes.object.isRequired,
-    },
+export default class MessageComposerInput extends React.Component {
+    constructor(props, context) {
```

Comment:
We've been sticking to non-js class react components, and we've stuck with this in newer code too since js classs components don't do automatic method binding, so here we probably should be consistent and use `React.createClass`.